### PR TITLE
TEE_{Create,Open}PersistentObject now panic when object is NULL

### PR DESCRIPTION
**Pull request by c-auger**
According to GP specification, passing NULL for object pointer should
panic, and not return an error.
Some harmonization has also been done in these two functions to use
the same conventions as surrounding code.

Signed-off-by: Cedric Auger <cedric.auger@provenrun.com>

### CODE REVIEW

**Comment by c-auger** [comment]
This pull request concerns issue #364 on optee_tests.

I could not test it, as I am using another backend than OPTEE, but I at least checked that this compiles.

Warning: testing it against optee_test will fail for some tests, as some optee_tests (at least until this is fixed) pass a NULL pointer to these functions.

**Comment by jforissier** [comment]
> TEE_{Create,Open}PersistentObject now panic when object is NULL

Please add parentheses after function names. Also, when possible it is better to spell out function names completely and exactly. It helps when doing "git log --grep=..." for instance.
How about: `TEE_CreatePersistentObject(), TEE_OpenPersistentObject(): panic if object is NULL`?

> According to GP specification [...]

Can you cite the name/version of the spec and the section number? I can't find a place where this case is explicitly mentioned. Perhaps just say that `TEE_ERROR_BAD_PARAMETERS` is not listed as a valid return code and a NULL `object` falls in the Panic Reasons.

> I could not test it, as I am using another backend than OPTEE, but I at least checked that this compiles.

Testing with QEMU is easy enough, it's basically just "make run" and you can run `xtest`. See https://optee.readthedocs.io/building/devices/qemu.html.

> Warning: testing it against optee_test will fail for some tests

Then we need a PR for the tests, too.

Thanks.

**Comment by c-auger** [comment]
> 
> > Warning: testing it against optee_test will fail for some tests
> 
> Then we need a PR for the tests, too.
> 
> Thanks.

Yes, I requested pulls in the wrong order. I should have started with tests.
Failure at test 6012 should be fixed with Pull Request #368 on optee-tests.

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome@forissier.org>`

You may want to wait for https://github.com/OP-TEE/optee_test/pull/368 to be merged before adding the review tags in order to make sure all is good (and save the planet by avoiding too many rebuilds :D)

**Comment by jforissier** [comment]
Travis (checkpatch) has noticed that the patch Author (`Cedric Auger <cauger@provenrun.com>`) is different from your `Signed-off-by:` tag (`Cedric Auger <cedric.auger@provenrun.com>`).

File: lib/libutee/tee_api_objects.c [hunk 1]
```diff
@@ -434,40 +432,40 @@ TEE_Result TEE_CreatePersistentObject(uint32_t storageID, const void *objectID,
 	TEE_Result res;
 	uint32_t obj;
 
+	if (!object)
+		TEE_Panic(TEE_ERROR_BAD_PARAMETERS);
```
Comment (etienne-lms) [inline lib/libutee/tee_api_objects.c]: Actually, this is not needed. The TA will panic from a segfault if pointer is null.
Suggest to remove. Despite my R-b tag.

**Comment by etienne-lms** [comment]
Also, for consistency with git history, prefer quite short comment, and starting with `libutee:` since its about, leaving (long) function name to message body, something like:
```
libutee: panic on null pointer on object opening` 

Change TEE_CreatePersistentObject() and TEE_OpenPersistentObject() 
to panic when passed an null reference pointer.

According to the TEE Internal Core API Specification 1.1 and 1.2,
sections 5.7.1 and 5.7.2, passing NULL for object pointer should panic,
and not return an error.

Some harmonization has also been done in these two functions to use
the same conventions as surrounding code.

Signed-off-by: ...
```

**Comment by jforissier** [comment]
@jenswi-linaro hope you're good with then change suggested by @etienne-lms, please thumb up if we can still apply your tag, thanks!

**Comment by jenswi-linaro** [comment]
Yes, it's good.

**Comment by jforissier** [comment]
Tag added, merged manually. Thanks.

**Pull request closed without merging**